numpy/core/tests/test_multiarray.py

```diff
@@ -6726,6 +6727,27 @@ def test_null_inside_ustring_array_is_truthy(self):
         a[0] = ' \0 \0'
         self.assertTrue(a)
 
+
+class TestCTypes(TestCase):
+
+    def test_ctypes_is_available(self):
+        test_arr = np.array([[1, 2, 3], [4, 5, 6]])
+
+        self.assertEqual(ctypes, test_arr.ctypes._ctypes)
+        assert_equal(test_arr.shape, (2, 3))
+
+    def test_ctypes_is_not_available(self):
+        _internal.ctypes = None
+        try:
+            test_arr = np.array([[1, 2, 3], [4, 5, 6]])
+
+            self.assertIsInstance(
+                test_arr.ctypes._ctypes, _internal._missing_ctypes)
```

Comment:
Hey Eric,

I tried again with the sys.modules hackery, and trying to put things back together again with the try/finally block. Unfortunately, other tests continue to fail. I guess this is because tests are run in parallel. With the current setup (setting ctypes = None), we will have similar cross-test side effects, but I guess we are getting lucky in that it isn't causing any errant failures.

By "spin off a new python interpreter", are you suggesting to execute the test with python's subprocess?

Eg (?):
def some_test(self):
    self.assertEqual(0, subprocess.call(sys.executable, "-c", test_code)